packages/babel-preset/src/assets/replaceAssetsWithImports.ts

```diff
@@ -40,10 +40,23 @@ export function replaceAssetsWithImports(
         }
 
         const result = parseStringWithUrl(value);
-        const assetIdentifier = getAssetIdentifier(result.url);
+        if (result.url.startsWith('data:') || result.url.startsWith('http:') || result.url.startsWith('https:')) {
+          return;
+        }
+
+        // Handle `filter: url(#id)` and `filter: url(./a.svg#id)`
+        const [pathname, hash] = result.url.split('#');
+        if (!pathname) {
+          return;
+        }
+
+        const assetIdentifier = getAssetIdentifier(pathname);
```

Comment:
Is it mandatory to do any changes there? We ran there on `StringLiteral` and it could be there only if an asset contains imports.

Reply:
Didn't `normalizeStyleRules.ts` output string literal in both case? As noted in OP, current version will transform `data:` URLs to assets:

Input:

```js
import { makeStyles } from "@griffel/core";

export const useClasses = makeStyles({
  root: {
    backgroundImage: `url(data:foo)`
  }
})
```

Output:

```js
import _asset from "../data:test";
import { __styles } from "@griffel/core";
export const useClasses = __styles({
  "root": {
    "Bcmaq0h": "f1tpm5in"
  }
}, {
  "d": [`.f1tpm5in{background-image:url(${_asset});}`]
});
```

Reply:
@yume-chan yeah, sorry for confusion. You're correct 👍